src/components/checkbox/checkbox.njk

```diff
@@ -1,30 +1,73 @@
-<legend class="rvt-sr-only">{{ legend }}</legend>
-<ul{% if list %} class="rvt-plain-list"{% else %} class="rvt-inline-list"{% endif %}>
-  {% for box in boxes %}
-    <li{% if box.hidden %} class="rvt-checkbox-wrapper"{% endif %}>
-      <input class="rvt-checkbox" type="checkbox" name="checkbox-{{ box.id }}" id="{{ box.id }}"{% if box.modifiers %}{% for modifier in box.modifiers %} {{ modifier.status }}{% endfor %}{% endif %}>
-      {% if box.hidden %}
+<fieldset class="rvt-fieldset">
+  <legend class="rvt-sr-only">Checkbox list</legend>
+  <ul class="rvt-plain-list rvt-width-xl">
+    <li>
+      <span class="rvt-checkbox">
+        <input class="rvt-checkbox__input" type="checkbox" name="checkbox-demo" id="checkbox-1">
+        <!--
+          Can work with hidden input elements that some framework template languages generate
+        -->
         <input type="hidden">
-      {% endif %}
-      <label class="rvt-label" for="{{ box.id }}">{{ box.checkboxLabel }}</label>
-      {% if (box.id === "checkbox-indeterminate") %}
-        <ul class="rvt-plain-list rvt-p-left-lg">
-          {% for item in box.items %}
-            <li>
-              <input class="rvt-checkbox" type="checkbox" name="checkbox-{{ item.id }}" id="{{ item.id }}"{% if loop.index === 1 %} checked{% endif %}>
-              <label class="rvt-label" for="{{ item.id }}">{{ item.checkboxLabel }}</label>
-            </li>
-          {% endfor %}
-        </ul>
-      {% endif %}
+        <label for="checkbox-1">Option one</label>
+      </span>
     </li>
-  {% endfor %}
+    <li>
+      <span class="rvt-checkbox">
+        <input type="checkbox" name="checkbox-demo" id="indeterminate">
+        <label for="indeterminate">Option two</label>
+      </span>
+    </li>
+    <li>
+      <span class="rvt-checkbox">
+        <input type="checkbox" name="checkbox-demo" id="checkbox-3" disabled>
+        <label for="checkbox-3">Option three</label>
+      </span>
+    </li>
+    <li>
+      <span class="rvt-checkbox">
+        <input type="checkbox" name="checkbox-demo" id="checkbox-4" disabled checked>
+        <label for="checkbox-4">Option four</label>
+      </span>
+    </li>
+    <li>
+      <span class="rvt-checkbox">
+        <input type="checkbox" name="checkbox-demo" id="checkbox-long">
+        <label for="checkbox-long">
+          <span class="rvt-text-bold">Just a quick note</span><br>
+          <span class="rvt-ts-14 rvt-display-inline-block rvt-color-black-600">This checkbox has a really long label that can wrap on to two lines and still have nice left alignment.</span>
+        </label>
+      </span>
+    </li>
+    <li>
+      <span class="rvt-checkbox">
+        <input type="checkbox" name="checkbox-demo" id="checkbox-hidden-label">
+        <label for="checkbox-hidden-label">
```

Comment:
This needs `class="rvt-p-all-none"` to make the hit area just the checkbox (24x24), instead of something bigger (40x24).

Alternatively, we could add a modifier (`.rvt-checkbox--sr-only`?) that could automate this and simplify the markup. See "Option ten" in the demo. This works well in Safari+VoiceOver. And it would work well in the situations where you can't modify the inner markup.

https://codepen.io/basham/pen/yLVmBRJ

```html
<div class="rvt-checkbox rvt-checkbox--sr-only">
  <input type="checkbox" name="checkbox-demo" id="checkbox-10" checked>
  <label for="checkbox-10">Option ten, screen reader only label</label>
</div>
```

```css
.rvt-checkbox--sr-only input[type=checkbox] + label {
  height: 0;
  overflow: hidden;
  padding: 0;
  width: 0;
}
```

Reply:
Maybe a better class name is `rvt-checkbox--sr-only-label`? Don't know. Needs workshopped.

Reply:
@basham I like the idea of a modifier on the checkbox wrapper element for the uncontrolled markup situations you mention. `rvt-checkbox--sr-only-label` more accurately describes what is going on, but it also feels really verbose. :(

I'm fine with using `rvt-checkbox--sr-only-label`. I'd rather error on the side of clarity.